clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy

```diff
@@ -361,5 +362,23 @@ class KubernetesClientApiAdapter {
       }
     }
   }
+
+  String getControllerType(String name, String namespace, String controllerType) {
```

Comment:
Why pass in the type to get the type? I'm not clear on what this is supposed to do

Reply:
This mainly to deal with same for different controller type.  we need to go through each type.